hooks/go-imports/go-imports.go

```diff
@@ -16,62 +18,102 @@ import (
 	"github.com/inturn/pre-commit-gobuild/internal/helpers"
 )
 
+type lintError struct {
+	err  error
+	path string
+}
+
 func main() {
 	workDir, err := os.Getwd()
 	if err != nil {
 		log.Fatal(err)
 	}
 	dirs := helpers.DirsWith(workDir, "\\.go$")
 
+	errc := make(chan lintError, 10)
 	wg := &sync.WaitGroup{}
 
-	for _, dir := range dirs {
-		if !strings.Contains(dir, "/vendor/") {
-			files, err := ioutil.ReadDir(dir)
-			if err != nil {
-				log.Printf("error occured on read dir %s: %s", dir, err)
-			}
-			for _, f := range files {
-				if !strings.HasSuffix(f.Name(), ".go") {
-					continue
+	go func() {
+		for _, dir := range dirs {
+			if !strings.Contains(dir, "/vendor/") {
+				files, err := ioutil.ReadDir(dir)
+				if err != nil {
+					log.Printf("error occured on read dir %s: %s", dir, err)
+				}
+				for _, f := range files {
+					if !strings.HasSuffix(f.Name(), ".go") {
+						continue
+					}
+					wg.Add(1)
+					go func(d, name string) {
+						sortFileImports(filepath.Join(d, name), errc)
+						wg.Done()
+					}(dir, f.Name())
 				}
-				wg.Add(1)
-				go func(d, name string) {
-					sortFileImports(filepath.Join(d, name))
-					wg.Done()
-				}(dir, f.Name())
 			}
 		}
+		wg.Wait()
+		close(errc)
+	}()
+
+	var le []lintError
+	for lintErr := range errc {
+		log.Println(lintErr.path, lintErr.err)
+		le = append(le, lintErr)
+	}
+	if len(le) != 0 {
```

Comment:
@yramanovich It seems it is a number of errors. It is because we may have some other error than "file has changed". I think we should print "errors count: " or something like.

Reply:
@guntenbein fixed